### PR TITLE
fix(go): support processing footers in gapics

### PR DESCRIPTION
**Pull request by codyoss**
Because we need to filter the commits found in the footer of a
commit we needed a hook to filter commits for the changelog. This
can't be done in the releaser as it only has the concept of raw
commits.Added the idea of supplying an optional commit filter to
ConventionalCommits.

### CODE REVIEW

File: src/conventional-commits.ts [hunk 1]
```diff
@@ -71,7 +72,10 @@ interface Note {
   text: string;
 }
 
-function getParsedCommits(commits: Commit[]): CommitWithHash[] {
+function getParsedCommits(
+  commits: Commit[],
+  commitFilter?: (c: ConventionalChangelogCommit) => boolean
```
Comment (bcoe) [inline src/conventional-commits.ts:77]: this is a clever approach.

File: src/conventional-commits.ts [hunk 2]
```diff
@@ -81,6 +85,9 @@ function getParsedCommits(commits: Commit[]): CommitWithHash[] {
         const commitWithHash = postProcessCommits(
           parsedCommit
         ) as CommitWithHash;
+        if (commitFilter?.(parsedCommit)) {
```
Comment (bcoe) [inline src/conventional-commits.ts:88]: what if we default the `commitFilter` to a filter that always returns `true`, the current behavior, then we don't need to use this optional call?

**Review by bcoe — CHANGES_REQUESTED** [review]
I like the approach you've come up with 👍 one thought, perhaps we could just default the filter to: `() => true`, then we don't need to do the optional call logic.

File: src/conventional-commits.ts (hunk 2 above)
Comment (codyoss) [reply src/conventional-commits.ts]: Defaulted to false, as that will allow a commit to be processed.

**Review by bcoe — APPROVED** [review]

**Pull request merged**